### PR TITLE
Map search query

### PR DESCRIPTION
**Pull request by desnoes**
Fix for [issue 1954](https://github.com/c2corg/v6_ui/issues/1954). The request is filtered to give results that correspond to the following OSM tags: city, town and village. This remove all street's adresses from the search result.

### CODE REVIEW

File: c2corg_ui/static/js/map/search.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ app.MapSearchController.SEARCH_URL = 'https://photon.komoot.de/api/';
 app.MapSearchController.prototype.createAndInitBloodhound_ = function() {
   let url = app.MapSearchController.SEARCH_URL;
   url += '?q=%QUERY';
+  url += '&osm_tag=place:city&osm_tag=place:town&osm_tag=place:village';
```
Comment (brunobesson) [inline c2corg_ui/static/js/map/search.js]: This should rather be written in a single line, with query in the end (although it's only cosmetics)

`url += '?osm_tag=place:city&osm_tag=place:town&osm_tag=place:village&q=%QUERY';`

**Comment by asaunier** [comment]
@desnoes Nice!
I wonder if your change is not a bit too restrictive though. I remember it was possible to search also for location names such as mountains (eg. "Chamechaude") or even regions "Valais". Don't you think it would be nice to keep such features?
Is there a page listing the available values for "osm_tag"?

In addition: do you really need 2 commits? :P 
Again it is better to tidy commits, see https://github.com/c2corg/v6_ui/wiki/Git-Cookbook#managing-commits

**Comment by desnoes** [comment]
@asaunier : it tested with key word 'mont-blanc' and i did not get any result, so i concluded summits were not returned. This was a mistake, i checked with Chamechaude and i get the result. I will try other tags.
The list of tags is given here: https://wiki.openstreetmap.org/wiki/Map_Features. There is an interesting list of tags for mountain activities in section 'Landform related'.

For the commit, this is an error from my side.

**Comment by desnoes** [comment]
I did some investigations. ThHe problem is not solved with a filer defined with osm_tags. I fact the bug comes from the location bias. 
If i comment the code related to search with priority to a geo position, then the search is working as expected. For instance, when i type 'Chambéry', the city is Rhône-Alpes is return in the first result, whereas it is not in the list with the current version. If i type 'chamechaude', the peak is sorted in the first position and streets containing this string are given after.
I don't see the need for a search with location bias in our map search and thus propose to comment these lines of code:

` const center = this.map.getView().getCenter();
  if (center !== undefined) {
     // give priority to nearby results
     const centerWgs84 = ol.proj.toLonLat(center);
     url += '&lon=' + centerWgs84[0] + '&lat=' + centerWgs84[1];
   }`

Do you agree ?

**Comment by asaunier** [comment]
@desnoes

>  add 'distance_sort=false' in the command line to solve the bug and i works very fine !

OK then, please merge.
Could you:
* remove the useless commit
* fix the typo in the other commit message. By the way prefer using explicit but simple commit message: think that they will appear in the commits log https://github.com/c2corg/v6_ui/commits/master => no need of "v2" etc.

**Comment by desnoes** [comment]
@asaunier : thanks for the review. I will remove the useless commit

>Just a question though: komoot/photon#297 makes distance_sort=false a default value in photon. Then do we really need to add this value explicitly? Perhaps we only have to wait photon service is updated with this new default value?

The default value is not yet applied by photon. I don't know when they will update their service. In the meantime, i think it is usefull to correct the bug.

**Comment by asaunier** [comment]
@desnoes Don't forget to delete the branch after merging the PR => "Delete branch" button above.

**Pull request merged**